### ISSUE TITLE
Incorrect _.bind when (ab?)using _.find with property shorthand and a thisArg

### ISSUE THREAD

**Issue by davidshepherd7** [jfmengels/lodash-codemods#2]
In 3.10 it was possible to use a three argument shorthand for find, such that this

    _.find(collection, 'id', x);

is equivalent in 4.13 to

    _.find(collection, ['id', x]);

However the codemod outputs

    _.find(collection, _.bind('id', x));

which fails. This "feature" doesn't seem to have been documented anywhere, so I think it's an unintended consequence of combining `thisArg` and `_.matchesProperty`.  I suspect the same thing  worked for all functions that could take a property shorthand and a `thisArg`.

But anyway it somehow it got used in a few places in our code base, so I thought you might like to know that the codemod doesn't deal with it. Maybe you could add a note in the readme, or even implement the migration?

**Comment by jdalton** [comment]
> Do you know of other functions that were like this in v3? I'm thinking of filter, reject for instance.

Any of the ones that processed their iteratees with `_.iteratee` (or the v3 equiv).

**Comment by davidshepherd7** [comment]
Yeah just handling the case where it's a hard coded string and adding a note in the docs is enough. To handle all cases perfectly you would need to evaluate arbitrary code until you find out whether the result is a string or a function, and that sounds very difficult!

Hopefully it's not something that was widely used since it's undocumented!

**Comment by eburi** [comment]
I got good results using this in lib/this-arg-removal.js:

```
  .forEach(function (p) {
    var thisPos = methodsThisPos[getName(p)];
    var args = p.value.arguments;
    if (args.length < thisPos) {
      return;
    }

    if(args[thisPos - 2].type == 'Literal') {
      p.value.arguments = args.slice(0, thisPos - 2).concat([
          j.arrayExpression([args[thisPos - 2], args[thisPos - 1]])
      ]);
    }
    else {
      p.value.arguments = args.slice(0, thisPos - 2).concat([
        j.callExpression(
            j.memberExpression(j.identifier('_'), j.identifier('bind')),
            [args[thisPos - 2], args[thisPos - 1]]
        )
      ]);
    }
  });

```
